lib/amazonka/src/Amazonka/Env.hs-boot

```diff
@@ -0,0 +1,16 @@
+module Amazonka.Env where
+
+import Amazonka.Types (Region, Logger, Service, Auth)
+import Amazonka.Prelude
+import qualified Network.HTTP.Client as Client
+import {-# SOURCE #-} Amazonka.Env.Hooks (Hooks)
+
+data Env' (withAuth :: Type -> Type) = Env
```

Comment:
Seems somewhat arbitrary to the naive reader (ie. me) parameterising over some `f :: Type -> Type` when you could just define:

```haskell
data EnvF auth = Env {
    ...,
    auth :: auth,
} deriving stock (Functor, ...)

type Env = EnvF (Identity Auth)
```

And pass parameters as: 

```haskell
retryRequest ::
  forall m a f.
  ( Foldable f
  ) =>
  EnvF (f Auth) ->
  a ->
  m (Either Error (ClientResponse (AWSResponse a)))
```

Reply:
Let's discuss in a separate issue, since the `Type -> Type` version has been in since the [env rework](https://github.com/brendanhay/amazonka/pull/746/files#diff-af09bf5533244a54c9ff45b3d0a799424cf4e39c48af0726e0cf5c5b1161ca96R54-R61) in #746 . It shows up in this diff because this PR added a `.hs-boot` file.

Reply:
Opened #877 - I dunno if/when github pings people.